### PR TITLE
docs: add v1 and v2 badges

### LINKED ISSUE #6328: Meta: Badging for Pydantic v1 and v2
It would be nice to have a badge for Pydantic to add them to repos, similar to Ruff's: https://github.com/astral-sh/ruff/blob/main/assets/badge/v2.json

Endpoints are documented here:
- https://shields.io/badges/dynamic-json-badge
- https://shields.io/badges/dynamic-yaml-badge

Examples:

[Pydantic](https://pydantic.dev)

What the finished product to advertise might look like (URLs need updating once merge, i think?):
[Pydantic v1](https://pydantic.dev)
[Pydantic v2](https://pydantic.dev)

Related Gist:

https://gist.github.com/JacobCoffee/482c716c1ee64bab9d073583c992503f

Selected Assignee: @lig

### PR DESCRIPTION
**Pull request by JacobCoffee**
## Change Summary

With the launch of Pydantic v2, new and repeat visitors alike will be looking to update their repos. This gives way to an opportunity for a little advertisement via badging - to indicate my repo is using Pydantic.

## Related issue number

Closes #6328 

## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Example v2 badge:

[Pydantic](https://pydantic.dev)

Related Gist:

https://gist.github.com/JacobCoffee/482c716c1ee64bab9d073583c992503f

What the finished product to advertise might look like (URLs need updating once merge, i think?):
[Pydantic v1](https://pydantic.dev)
[Pydantic v2](https://pydantic.dev)

Selected Reviewer: @samuelcolvin

(skip change file check)

### CODE REVIEW

**Comment by JacobCoffee** [comment]
please review

Not sure if you would want the v1 badge to be reflective of the old logo?

**Comment by lig** [comment]
Personally, I see no rationale in these badges. Ruff, black, mypy, etc show contributors the way a codebase is organized. I understand that there are projects that heavily rely on Pydantic and it's not a simple dependency for such projects. However, at the end of the day Pydantic is just a library a project depend on. Such badge says nothing about the way code is written to a contributor.

**Comment by JacobCoffee** [comment]
I see what you mean! But i was thinking it would be particularly nice for frameworks that utilize Pydantic heavily, maybe.

**Comment by samuelcolvin** [comment]
Seems like a nice idea to me. No idea how much people will use it, but no harm in trying.

Please could you:
* add a section to `contributing.md` explaining how to use the badges
* change the color behind the "v1" or "v2" to green - I'm assuming you've used the pydantic pink, which makes sense but makes it look (to me who's partially color blind) a bit like "red for failuing"

please update.

**Comment by JacobCoffee** [comment]
I think this is everything, completed - let me know if i missed something 😄

**Comment by lig** [comment]
please review

File: docs/contributing.md [hunk 1]
```diff
@@ -90,8 +89,8 @@ Run tests and linting locally to make sure everything is working as expected.
 ```bash
 # Run automated code formatting and linting
 make format
-# Pydantic uses black and ruff
-# (https://github.com/ambv/black, https://github.com/charliermarsh/ruff)
+# Pydantic uses black and pydantic
+# (https://github.com/ambv/black, https://github.com/charliermarsh/pydantic)
```
Comment (samuelcolvin) [inline docs/contributing.md:93]:
```suggestion
# Pydantic uses black and ruff
# (https://github.com/ambv/black, https://github.com/charliermarsh/ruff)
```

File: README.md [hunk 2]
```diff
@@ -7,6 +7,7 @@
 [![downloads](https://pepy.tech/badge/pydantic/month)](https://pepy.tech/project/pydantic)
 [![versions](https://img.shields.io/pypi/pyversions/pydantic.svg)](https://github.com/pydantic/pydantic)
 [![license](https://img.shields.io/github/license/pydantic/pydantic.svg)](https://github.com/pydantic/pydantic/blob/main/LICENSE)
+[![Pydantic v2](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/pydantic/pydantic/5697b1e4c4a9790ece607654e6c02a160620c7e1/docs/badge/v2.json)](https://pydantic.dev)
```
Comment (lig) [inline README.md:10]: @samuelcolvin I'm not a fan of having the badge on the Pydantic's own README.

**Review by lig — APPROVED** [review]
I'm approving to pass the final decision to @samuelcolvin .

While I can understand the rationale behind having this badge for a project depending on Pydantic, I don't see what the badge is telling users on Pydantic's own readme.

**Comment by samuelcolvin** [comment]
Thanks so much.

**Pull request merged**